evap/staff/importers/json.py

```diff
@@ -0,0 +1,325 @@
+            log += f"- {new_evaluation}\n"
+        log += self._make_total(len(self.new_evaluations))
+
+        log += self._make_heading("Updated Courses")
+        for updated_course in self.updated_courses:
+            log += f"- {updated_course}\n"
+        log += self._make_total(len(self.updated_courses))
+
+        log += self._make_heading("Updated Evaluations")
+        for updated_evaluation in self.updated_evaluations:
+            log += f"- {updated_evaluation}\n"
+        log += self._make_total(len(self.updated_evaluations))
+
+        log += self._make_heading("Attempted Changes")
+        for attempted_change in self.attempted_changes:
+            log += f"- {attempted_change}\n"
+        log += self._make_total(len(self.attempted_changes))
+
+        return log
+
+
+class JSONImporter:
+    DATETIME_FORMAT = "%d.%m.%Y %H:%M"
+
+    def __init__(self, semester: Semester) -> None:
+        self.semester = semester
+        self.user_profile_map: dict[str, UserProfile] = {}
+        self.course_type_cache: dict[str, CourseType] = {}
+        self.degree_cache: dict[str, Degree] = {}
+        self.course_map: dict[str, Course] = {}
+        self.statistics = ImportStatistics()
+
+    def _get_course_type(self, name: str) -> CourseType:
+        if name in self.course_type_cache:
+            return self.course_type_cache[name]
+
+        course_type = CourseType.objects.get_or_create(name_de=name, defaults={"name_en": name})[0]
+        self.course_type_cache[name] = course_type
+        return course_type
+
+    def _get_degree(self, name: str) -> Degree:
+        if name in self.degree_cache:
+            return self.degree_cache[name]
+
+        degree = Degree.objects.get_or_create(name_de=name, defaults={"name_en": name})[0]
+        self.degree_cache[name] = degree
+        return degree
+
+    def _get_user_profiles(self, data: list[ImportRelated]) -> list[UserProfile]:
+        return [self.user_profile_map[related["gguid"]] for related in data]
+
+    def _import_students(self, data: list[ImportStudent]) -> None:
+        for entry in data:
+            email = clean_email(entry["email"])
+            user_profile, __, changes = update_or_create_with_changes(
+                UserProfile,
+                email=email,
+                defaults={"last_name": entry["name"], "first_name_given": entry["christianname"]},
+            )
+            user_profile: UserProfile
+            if changes:
+                change = NameChange(
+                    old_last_name=changes["last_name"][0] if changes.get("last_name") else user_profile.last_name,
+                    old_first_name_given=(
+                        changes["first_name_given"][0]
+                        if changes.get("first_name_given")
+                        else user_profile.first_name_given
+                    ),
+                    new_last_name=changes["last_name"][1] if changes.get("last_name") else user_profile.last_name,
```

Comment:
```suggestion
                    new_last_name=user_profile.last_name,
```
and so on?

(To me, it also makes sense to only create a `NameChange` instance if there was actually a name change, then we also don't need the `changes["last_name"][0] if changes.get("last_name") else user_profile.last_name` part above)